src/SplitTreatments.tsx

```diff
@@ -1,41 +1,46 @@
 import React from 'react';
+import memoizeOne from 'memoize-one';
 import shallowEqual from 'shallowequal';
 import SplitContext from './SplitContext';
 import { ISplitTreatmentsProps, ISplitContextValues } from './types';
 import { getControlTreatmentsWithConfig, WARN_ST_NO_CLIENT } from './constants';
 
+function argsAreEqual(newArgs: any[], lastArgs: any[]): boolean {
+  return newArgs[0] === lastArgs[0] && // client
```

Comment:
Is it expected that the client might change?

Reply:
yes. A higher `SplitClient` or SplitFactory` component in the component tree might change the client at the context, which might be a new client for the same factory or even a client from a different factory, but there is no need to compare the factory.